liblwgeom/lwgeom_geos.c

```diff
@@ -1223,13 +1227,97 @@ lwgeom_offsetcurve(const LWLINE* lwline, double size, int quadsegs, int joinStyl
 
 	g3 = GEOSOffsetCurve(g1, size, quadsegs, joinStyle, mitreLimit);
 
-	if (!g3) return geos_clean_and_fail(g1, NULL, NULL, __func__);
+	if (!g3)
+	{
+		geos_clean(g1, NULL, NULL);
+		return NULL;
+	}
 
 	if (!output_geos_as_lwgeom(&g3, &result, srid, is3d, __func__))
 		return geos_clean_and_fail(g1, NULL, g3, __func__);
 
 	geos_clean(g1, NULL, g3);
+	return result;
+}
+
+static LWGEOM*
+lwmline_offsetcurve(const LWGEOM* geom, double size, int quadsegs, int joinStyle, double mitreLimit)
+{
+	int32_t srid = get_result_srid(geom, NULL, __func__);
+	uint8_t is3d = FLAGS_GET_Z(geom->flags);
+	if (srid == SRID_INVALID) return NULL;
+
+	if (geom->type == LINETYPE)
```

Comment:
The rest of the codebase uses case statements instead of if-defs. Please, consider switching it

Reply:
switched to case.